src/lib/logical_query_plan/lqp_expression.cpp

```diff
@@ -47,6 +53,11 @@ void LQPExpression::set_column_reference(const LQPColumnReference& column_refere
   _column_reference = column_reference;
 }
 
+std::shared_ptr<AbstractLQPNode> LQPExpression::subselect_node() const {
+  DebugAssert(_subselect_node != std::nullopt, "LPQNode does not contain a subselect node.");
```

Comment:
I'm in favor of 

    DebugAssert(_subselect_node, "LPQNode does not contain a subselect node.");

instead of `!= std::nullopt` or in case of (smart) pointers `!= nullptr`.

Reply:
This should probably not be a  `DebugAssert` because it is an invariant that should always hold.

Reply:
This is not so much about if it should always be true but more about "we reeeeally don't expect this to happen, but we still want to write down our assertions and have them fail when we're debugging things."